src/credentials/FabricTable.cpp

```diff
@@ -360,6 +360,53 @@ CHIP_ERROR FabricInfo::SetExternallyOwnedOperationalKeypair(P256Keypair * keyPai
     return CHIP_NO_ERROR;
 }
 
+CHIP_ERROR FabricInfo::ValidateNewNOCCredentials()
+{
+    P256PublicKey pubkey;
+    ValidationContext validContext;
+    validContext.Reset();
+    validContext.mRequiredKeyUsages.Set(KeyUsageFlags::kDigitalSignature);
+    validContext.mRequiredKeyPurposes.Set(KeyPurposeFlags::kServerAuth);
+
+    // Make sure to not modify any of our state until VerifyCredentials passes.
```

Comment:
Actually, given how this is being used now, on a FabricInfo that already has the new NOC/etc, it's not really an issue if we _do_ modify other members.

Reply:
removed this comment